prediction/src/tests/test_calculate_volume.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import os
+import pytest
+import shutil
+
+from ..algorithms.segment import trained_model
+
+
+@pytest.fixture
+def generate_motes(mask, centroid, volume):
+    centroid_ = np.asarray([centroid['x'], centroid['y'], centroid['z']])
+    free_voxels = np.where(mask != -1)
+    free_voxels = np.asarray(free_voxels).T
+    free_voxels = sorted(free_voxels, key=lambda x: np.linalg.norm(x - centroid_, ord=2))
+    free_voxels = np.asarray(free_voxels[:volume]).T
+    mask[(free_voxels[0], free_voxels[1], free_voxels[2])] = True
+
+
+def generate_mask(shape, centroids, volumes):
+    mask = np.zeros(shape, dtype=np.bool_)
+    for centroid, volume in zip(centroids, volumes):
+        generate_motes(mask, centroid, volume)
+    return mask
+
+
+def test_calculate_volume_on_unoverlapped_connected_components():
+    centroids = [[0, 0, 0], [32, 32, 28], [45, 45, 12]]
+    centroids = [{'x': centroid[0], 'y': centroid[1], 'z': centroid[2]} for centroid in centroids]
+    mask = generate_mask(shape=[50, 50, 29], centroids=centroids, volumes=[100, 20, 30])
+    assert mask.sum() == 150
+
+    path = os.path.join('.', 'tmp', 'mask.npy')
```

Comment:
Please use pylint's ``tmpdir`` support; it will — at the very least — clean up after itself on failing tests and avoid the need for ugly path manipulation!

Reply:
Thank you for pointing it out, I've fixed it.